### PR TITLE
Add genai-microservice-connectors(GMC) into GenAIInfra

### PR DESCRIPTION
**Pull request by zhlsunshine**
## Description

Add genai-microservice-connectors(GMC) which includes:

1. The CRD definition, including YAML file and golang source code 
2. The Key components of GMC: gmc controller and router
3. The templates yaml files of GMC
4. Unit test ...... 

## Issues

 `n/a`

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

golang, client-go

## Tests

unit test

### CODE REVIEW

**Comment by mkbhanda** [comment]
@hshen14 would you kindly approve. All checks passed.

**Comment by mkbhanda** [comment]
Let me approve given that @hshen14 and @chensuyue did review an earlier version and all tests passed.

**Review by mkbhanda — APPROVED** [review]
Reviewed by @hshen14 and @chensuyue earlier and all tests passing.

File: microservices-connector/test/e2e/e2e_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var _ = Describe("controller", Ordered, func() {
 		_, _ = utils.Run(cmd)
 	})
 
-	AfterAll(func() {
+	after all(func() {
```
Comment (zhlsunshine) [inline microservices-connector/test/e2e/e2e_test.go:33]: There is a bug imported by `[pre-commit.ci] auto fixes from pre-commit.com hooks`

**Comment by poussa** [comment]
> Reviewed by @hshen14 and @chensuyue earlier and all tests passing.

Where is this review?

**Pull request merged**